src/utils/string.js

```diff
@@ -4,21 +4,20 @@
 
 // @flow
 
-/**
- * Takes a string and returns the string with public URLs removed.
- * It doesn't remove the URLs like `chrome://..` because they are internal URLs
- * and they shouldn't be removed.
- */
-export function removeURLs(
-  string: string,
-  removeExtensions: boolean = true,
-  redactedText: string = '<URL>'
-): string {
-  const protocols = ['http', 'https', 'ftp', 'file', 'moz-page-thumb'];
-  if (removeExtensions) {
-    protocols.push('moz-extension');
-  }
-  const regExp = new RegExp(
+// Initializing this RegExp outside of removeURLs because that function is in a
+// hot path during sanitization and it's good to avoid the initialization of the
+// RegExp which is costly.
+const REMOVE_URLS_REGEXP = (function() {
+  const protocols = [
+    'http',
+    'https',
+    'ftp',
+    'file',
+    'moz-extension',
+    'moz-page-thumb',
+  ];
+
+  return new RegExp(
     `\\b((?:${protocols.join('|')})://)/?[^\\s/$.?#][^\\s)]*`,
```

Comment:
BTW I wonder how we behave for URLs that contain a user and password in the URL, for example:

https://user:password@website.com

(of course not for this PR)

Reply:
We do have some tests for them :)
https://github.com/firefox-devtools/profiler/blob/173c32a9d5d898d219de66a7be43c8bdeb518d21/src/test/unit/string.test.js#L35-L40